pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -584,24 +590,44 @@ func newHTTPPassThroughFilterChain(configgen *ConfigGeneratorImpl,
 	return filterChains
 }
 
-func newTCPProxyOutboundListenerFilter(push *model.PushContext, node *model.Proxy) *listener.Filter {
+func buildFallthroughNetworkFilters(push *model.PushContext, node *model.Proxy) []*listener.Filter {
 	tcpProxy := &tcp_proxy.TcpProxy{
 		StatPrefix:       util.BlackHoleCluster,
 		ClusterSpecifier: &tcp_proxy.TcpProxy_Cluster{Cluster: util.BlackHoleCluster},
 	}
+	requireSniForwarding := false
```

Comment:
So the forward via proxy functionality will be more debuggable if you add the same support for http and TLS sniffing in the virtual outbound. this way, you can atleast log the unknown traffic on the sidecar before forwarding it to the egress gateway. It doesn't have to be in this PR though.